lib/preview-web/src/renderDocs.tsx

```diff
@@ -35,9 +39,11 @@ async function renderDocsAsync<TFramework extends AnyFramework>(
   // Use `componentId` as a key so that we force a re-render every time
   // we switch components
   const docsElement = (
-    <DocsContainer key={story.componentId} context={docsContext}>
-      <Page />
-    </DocsContainer>
+    <CacheProvider value={emotionCache}>
```

Comment:
I've decided to wrap this here, even though theme (and other things too) is provided in `DocsContainer` because I think that it's best not to bust the cache on this `key` change. In reality Emotion's cache can't be as easily busted and recreated (it's not impossible but I don't see why I would want to implement this here) so you'd end up with a lot of duplicated styles with each new `key` on the `DocsContainer`

Reply:
@Andarist I'm not quite following:

> perhaps this is not the best place to do this since DocsContainer is totally configurable and is being injected here

Why is that an issue? Are you thinking that maybe people might want a custom container that doesn't use this emotion cache business?

BTW on the subject of the `key={}` there -- it looks like it was added for the sake of angular (legacy) inline rendering: https://github.com/storybookjs/storybook/pull/16149

So perhaps we could drop it in 7.0 (as we are dropping the old inline rendering).

Reply:
@Andarist having chatted with others, seeing as we are getting rid of the old inline rendering, the use of `key={}` here is no longer needed. With this in mind would you change the placement of the `<CacheProvider>`?

Reply:
I'm happy to update the PR with both changes (dropping the key, and moving the provider) if you let me know!

Reply:
Let's maybe wait a week or so before moving on with this PR - I will try to investigate this in the Emotion itself.

Reply:
A little bit later than I've promised - but I took a look there and I think that I understand where Stylis puts the comments much better now. Gonna recheck a couple of things but I think that I'm close to fixing this.

Reply:
< 2x the promised time is pretty darn good in my books!